src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -673,18 +712,21 @@ public async Task<IList<Check>> GetPullRequestChecksAsync(string pullRequestUrl)
                     }
                     catch (Exception ex) when (ex is HttpRequestException || ex is TaskCanceledException)
                     {
-                        if (retryCount <= 0)
+                        if (retriesRemaining <= 0)
                         {
-                            logger.LogError($"Failed to send HttpRequest to Azure DevOps API after {maxAttempts} attempts. Path: {requestPath}. Exception: {ex.ToString()}");
+                            if (logFailure)
+                            {
+                                logger.LogError($"Failed to send HttpRequest to Azure DevOps API after {retriesRemaining} attempts. Path: {requestPath}. Exception: {ex.ToString()}");
+                            }
                             throw;
                         }
-                        else
+                        else if (logFailure)
                         {
-                            logger.LogWarning($"Failed to send HttpRequest to Azure DevOps API: Path:{requestPath}. {retryCount} attempts remaining. Exception: {ex.ToString()}");
+                            logger.LogWarning($"Failed to send HttpRequest to Azure DevOps API: Path:{requestPath}. {retriesRemaining} attempts remaining. Exception: {ex.ToString()}");
                         }
                     }
-                    --retryCount;
-                    int delay = (maxAttempts - retryCount) * rng.Next(1, 7);
+                    --retriesRemaining;
+                    int delay = (retriesRemaining - retriesRemaining) * rng.Next(1, 7);
```

Comment:
A - A is zero. This is broken, and will just spam all 15 requests immediately.  I think one of these is supposed to be retryCount?

Reply:
I think now it's supposed to say "retryCount - retriesRemaining"?

Reply:
Woops, refactor fail.